enterprise/views.py

```diff
@@ -106,12 +122,54 @@ def get_note(provider, required):
                 " not be able to get any discounts from {provider}."
             ).format(provider=provider)
 
-    def get(self, request):
+    @staticmethod
+    def get_course_note(provider, course_name):
         """
-        Render a form to collect user input about data sharing consent.
+        Get a course-specific note for the form.
+
+        Arguments:
+            provider: The name of the linked EnterpriseCustomer
+            course_name: The name of the course in question
+        """
+        return _(
+            "Data sharing consent is required for courses provided by {provider}, including "
+            "{course_name}; if consent is not provided, you will be returned to the Dashboard."
+        ).format(provider=provider, course_name=course_name)
+
+    def get_course_specific_consent(self, request, course_id):
+        """
+        Render a form with course-specific information about data sharing consent.
+        """
+        try:
+            client = CourseApiClient()
+            course_details = client.get_course_details(course_id)
+        except Exception:
+            raise Http404
+        next_url = request.GET.get('next')
+
+        customer = get_object_or_404(EnterpriseCustomerUser, user_id=request.user.id).enterprise_customer
+        if not consent_necessary_for_course(request.user, course_id):
+            raise Exception
+        platform_name = configuration_helpers.get_value("PLATFORM_NAME", settings.PLATFORM_NAME)
+        course_name = course_details['name']
+        data = {
+            'platform_name': platform_name,
+            'data_sharing_consent': 'required',
```

Comment:
I believe we'll only be rendering this view when consent is required, but it wouldn't hurt to change.